### ISSUE TITLE
Make it easier to buy units and buildings from production menu

### ISSUE THREAD

**Issue by reijin90** [yairm210/UnCiv#417]
Right now, when one wants to buy units or buildings in the city they have to: 
Select the thing, click "build" or "produce" (or similar), after it is selected for production they can click on "Buy". 

How about adding a buy button to the select screen of a unit/building in the city? 

I'm guessing you could add the logic in the https://github.com/yairm210/UnCiv/blob/master/core/src/com/unciv/ui/pickerscreens/ConstructionPickerScreen.kt

similar to:

``` diff
for (building in GameBasics.Buildings.values) {
    if (!building.isBuildable(cityConstructions) && building.name!=cityConstructions.currentConstruction) continue
    val productionTextButton = getProductionButton(building.name,
            building.name + "\r\n" + cityConstructions.turnsToConstruction(building.name) + " {turns}".tr(),
            building.getDescription(true, currentPlayerCiv.policies.getAdoptedPolicies()),
            "Build [${building.name}]".tr())
    if (building.isWonder)
        wonders.addActor(productionTextButton)
    else
+        if (building !is SpecialConstruction &&
+                !(building is Building && building.isWonder)) {
+            row()
+            val buildingGoldCost = building.getGoldCost(city.civInfo.policies.getAdoptedPolicies())
+            val buildingBuyButton = TextButton("Buy for [$buildingGoldCost] gold".tr(), +CameraStageBaseScreen.skin)
+            buildingBuyButton.onClick("coin") {
+                city.cityConstructions.purchaseBuilding(city.cityConstructions.currentConstruction)
+                update()
+            }
+            if (buildingGoldCost > city.civInfo.gold) {
+                buildingBuyButton.disable()
+            }
            regularBuildings.addActor(productionTextButton)
+            regularBuildings.addActor(buildingBuyButton)
+        }
}

for (unit in GameBasics.Units.values.filter { it.isBuildable(cityConstructions)}) {
+    if (unit !is SpecialConstruction &&
+            !(unit is Building && unit.isWonder)) {
+        row()
+        val buildingGoldCost = unit.getGoldCost(city.civInfo.policies.getAdoptedPolicies())
+        val buildingBuyButton = TextButton("Buy for [$buildingGoldCost] gold".tr(), +CameraStageBaseScreen.skin)
+        buildingBuyButton.onClick("coin") {
+            city.cityConstructions.purchaseBuilding(city.cityConstructions.currentConstruction)
+            update()
+        }
+        if (buildingGoldCost > city.civInfo.gold) {
+            buildingBuyButton.disable()
+        }
        units.addActor(getProductionButton(unit.name,
            unit.name + "\r\n" + cityConstructions.turnsToConstruction(unit.name) + " {turns}".tr(),
            unit.getDescription(true), "Train [${unit.name}]".tr()))
+        regularBuildings.addActor(buildingBuyButton)
+    }
}
```

I'm aware this doesn't work as of now, especially because of this line: 
`city.cityConstructions.purchaseBuilding(city.cityConstructions.currentConstruction)`

and I'm not sure what the difference between add() and addActor() is, but I hope my idea comes across and my code is semi-usable

Best and keep it up! 👍

**Comment by yairm210** [comment]
So instead, I decided to remodel the entire left side of the city screen to look like the original civ.

Should enter 2.12.2 =D

Sneak peek:
![image](https://user-images.githubusercontent.com/8366208/51432661-e0b73380-1c43-11e9-9fc4-60b59b2972b1.png)

**Comment by reijin90** [comment]
nice! Also: Did you receive my email with the screenshots? Maybe it ended up in spam due to G-Drive links.

**Comment by yairm210** [comment]
No, I did not :|

**Comment by zivvel** [comment]
After this update, I no longer get tile info within city screen in the lower right corner. This means I can no longer buy tiles or acquire them from nearby cities.

**Comment by reijin90** [comment]
Confirmed. Also the translations are not applied to the items from the
scrollable list.

**Comment by yairm210** [comment]
@zivvel @reijin90  Good points!
Fixed, and will enter 2.12.4

**Comment by zivvel** [comment]
Overall it looks better and cleaner, but it's not as easy to see all the available production options at a glance.

After this update, we no longer have a way to see how many gears are stored & needed for current production target, like we do with food, culture.
